### ISSUE TITLE
Calculations for State Costs Summary table

### ISSUE THREAD

**Issue by pphillips-fearless** [18F/cms-hitech-apd#2169]
To prepare for the new "State Costs Summary" table, update the budget reducer math to capture "other funding amount" for the "state staff" and "non personnel costs" sections for each activity

### This task is done when...
 - [ ] the budget reducer code is updated to include "other funding amount" for the "state staff" and "non personnel costs" sections for each activity
 
this is a follow up to task #1937

**Comment by radavis** [comment]
Looking at the tests for the budget reducer. It looks like the data for ‘other funding’ is stored in costAllocation (see web/src/reducers/budget.test.js:73). I’m not 100% sure what this task is asking me to do. Do you want me to move this information somewhere else in the apd state?

**Comment by radavis** [comment]
@mgwalker Could you clarify what this statement means:

'the budget reducer code is updated to include "other funding amount" for the "state staff" and "non personnel costs" sections for each activity'

I think I know that the budget reducer takes in the APD state, and applies some math to it, creating totals. Is this task asking me to total the other funding amount?

Maybe I'm having difficulty understanding what is being asked for, here due to the fact that these tickets are ordered "back to front" instead of "front to back".

**Comment by pphillips-fearless** [comment]
@radavis I just put a little thought into this.  In the FFP and budget tables we have years[ffy].otherFunding
and on the mural at 
https://app.mural.co/t/gsa6/m/gsa6/1585942490744/d53ca4663eff051ca34dada15dc3d0172474eec8 
it shows the purple arrows bringing the amount over to the new table with a note that says "This is a scaled amount calculated by the app".  The purple arrows indicate that the other funding amount is split between State Staff, Non Personnel Costs, and Private Contractor Costs.  So we have a single number and we need it to be split into 3 numbers for use in the new table.  @mgwalker  does that align with what you were thinking?

**Comment by radavis** [comment]
Currently, the application allows you to enter an “Other Funding Amount” per activity, per year. How does the app determine how to split that value over “State Staff” and “Non-personnel Costs”? Is that logic that needs to be added as a part of this story? If so, how is the value split? If not, where are the values located in the application’s state object?

**Comment by beparticular** [comment]
@jeromeleecms I believe this is related to a conversation we were having about whether we need to surface the split in the app, based on is it a federal need, or is it a state need, or is it a need at all. IIRC we were going to talk about this more after our Cost Allocation team learning?

**Comment by jeromeleecms** [comment]
Let's talk through this during parking lot time -- this conversation is starting to feel like we are blurring the line between two separate areas... but let's talk through it to confirm.

**Comment by radavis** [comment]
Jerome provided me with this key piece of information on Friday:

![Screen Shot 2020-05-18 at 1.14.05 PM.png](https://images.zenhubusercontent.com/5e62a052e57324bfda300f92/d0598743-bcf8-460d-ad25-9cb8b1d967ca)

from: https://app.mural.co/t/eapd5455/m/eapd5455/1588711183141/acecb4562692e2b1e5bd412411b4d5c540d07b3e

**Comment by jeromeleecms** [comment]
Also to document the outcome of our parking lot time:

1) This would be need by both Federal and State users. There currently exists a "math gap" when we move from the summed budget for an activity -> the Medicaid share. By documenting what the other funding amount is removed from each line item, there will be no manual math that would need to occur in order for a state or federal review to establish what the Medicaid share was.

2) Also glad we clarified this! I thought we might be talking about cost allocation between different agencies, but, as we discussed, this is specifically only in the context of the three items (State staff, non-personnel costs, and private contractors). We'll def. talk about cost allocation in the broader context, but that is def. out of scope for this particular issue.
